DSN/backend/src/main/java/org/gfa/dsn/service/UserDetailsServiceImpl.java

```diff
@@ -25,12 +27,12 @@ public UserDetailsServiceImpl(UserRepository userRepository) {
     @Override
     public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
         logger.debug("Entering in loadUserByUsername Method...");
-        User user = userRepository.findByUsername(username);
-        if (user == null) {
+        Optional<User> user = userRepository.findByUsername(username);
+        if (user.isEmpty()) {
             logger.error("Username not found: " + username);
             throw new UsernameNotFoundException("could not found user..!!");
```

Comment:
I would create a new custom exception for this one, because it is connected to our business logic.
For example: **UserNotFoundException**

You can create a new folder for the custom exceptions.

Reply:
I approached an issue by trying to implement custom exception. Looks like an interface UserDetails doesn't want me to throw an error, cause it's not defined there
![image](https://github.com/MatejFrnka/social-network/assets/122916872/917f6a8f-a661-4831-96a9-b076132f7e0b)

Reply:
I just pushed a commit with custom exception